src/librustdoc/html/render/cache.rs

```diff
@@ -327,6 +327,13 @@ impl DocFolder for Cache {
                             parent_idx: None,
                             search_type: get_index_search_type(&item),
                         });
+
+                        for alias in item.attrs.get_doc_aliases().into_iter() {
+                            self.aliases
+                                .entry(alias.to_lowercase())
```

Comment:
No, because otherwise, we might have conflicts to handle on the JS side if we have `Hello` and `hello`. Also, I think it's better to lighten JS computations. In any case, we'll have to turn keys into lowercase, so I think it's better to do it directly on the Rust side once and for all.

Reply:
I guess it's not a big issue for now. It does seem weird that aliases are treated differently to the other search results though.

Reply:
The alias match is "exact", this is the base difference. But we can debate it in an issue if you want?